include/BSMPT/bounce_solution/action_calculation.h

```diff
@@ -0,0 +1,821 @@
+   * @return std::vector<double>
+   */
+  std::vector<double> ExactSolutionFromMinimum(double l0, double l);
+
+  /**
+   * @brief Calculates the 1D solution by comparing the @ref
+   * ExactSolutionCons and @ref ExactSolutionLin so that the analytical step is
+   * appropriate.
+   *
+   * @param l0 starting point
+   * @return std::vector<double>
+   */
+  std::vector<double> ExactSolution(double l0);
+
+  /**
+   * @brief This is done to make sure that we can still find solution after path
+   * deformation. This propagates the spline into negative values (by
+   * extrapolating).
+   *
+   * @return double negative or zero value.
+   */
+  double BackwardsPropagation();
+
+  /**
+   * @brief Integrates 1D bounce equation once
+   *
+   * @param l0 is the starting value.
+   * @param conv checks type of convergence. 0 = converged. -1 = undershot. +1 =
+   * overshot.
+   * @param rho vector of integration variable \f$ \rho \f$ steps.
+   * @param l vector of variable \f$ l \f$ steps.
+   * @param dl_drho vector of variable \f$ \frac{dl}{d\rho} \f$ steps.
+   * @param d2l_drho2 vector of variable \f$ \frac{d^2l}{d\rho^2} \f$ steps.
+   * @param maxiter is the maximum integration steps.
+   * @param error is the acceptance of undershoot/overshoot.
+   * @param eps_abs is used to control the step size error (RK4 vs RK5).
+   * @param max_step in the case you want to set a maximum step size in \f$ \rho
+   * \f$.
+   */
+  void IntegrateBounce(double l0,
+                       int &conv,
+                       std::vector<double> &rho,
+                       std::vector<double> &l,
+                       std::vector<double> &dl_drho,
+                       std::vector<double> &d2l_drho2,
+                       int maxiter,
+                       double error,
+                       double eps_abs,
+                       double max_step = 0);
+  /**
+   * @brief Performs a binary search using the overshooting/undershooting method
+   * to find the solution to the 1D bounce equation
+   *
+   * @param rho is the vector of \f$ \rho \f$ values from the solution
+   * @param l is the vector of \f$ l \f$ values from the solution
+   * @param dl_drho is the vector of \f$ \frac{dl}{d\rho} \f$ values from the
+   * solution
+   * @param d2l_drho2is the vector of \f$ \frac{d^2l}{d\rho^2} \f$ values from
+   * the solution
+   * @param error is the @ref IntegrateBounce error (acceptance of
+   * undershoot/overshoot)
+   * @param maxiter is the maximum number of binary searches
+   */
+  void Solve1DBounce(std::vector<double> &rho,
+                     std::vector<double> &l,
+                     std::vector<double> &dl_drho,
+                     std::vector<double> &d2l_drho2,
+                     double error = 1e-7,
+                     int maxiter  = 100);
+  /**
+   * @brief Formats a number to be written always with 3 characters, e.g. 1 ->
+   * 001, 19 -> 019
+   *
+   * @param number
+   * @return std::string
+   */
```

Comment:
What happens if I enter 1000?

Reply:
1000 would be printed, it just pads the numbers to the left if they are 0. 
Removed because it was only used when exporting the path deformation iterations (which was also removed).